tests/test_coreg.py

```diff
@@ -505,7 +505,20 @@ def test_blockwise_coreg_large_gaps(self):
         # Statistics are only calculated on finite values, so all of these should be finite as well.
         assert np.all(np.isfinite(stats))
 
+        # Copy the TBA DEM and set a square portion to nodata
+        tba = self.tba.copy()
+        tba.data[0, 450:500, 450:500] = -9999
+        tba.set_ndv(-9999)
 
+        blockwise = xdem.coreg.BlockwiseCoreg(xdem.coreg.NuthKaab(), 8, warn_failures=False)
+
+        # Align the DEM and apply the blockwise to a zero-array (to get the zshift)
+        aligned = blockwise.fit(self.ref, tba).apply(tba)
+        zshift = blockwise.apply(np.zeros_like(tba.data), transform=tba.transform)
+
+        # Validate that the zshift is not something crazy high and that no negative values exist in the data.
+        assert np.nanmax(np.abs(zshift)) < 50
+        assert np.count_nonzero(aligned.data.filled(0) < -50) == 0
```

Comment:
I've seen you use 'filled' quite often. I believe your intention is to exclude nodata from the statistics. If so, I recently found out you can use data.compressed(), which returns only the valid values as a 1D array (I know, the name is not very intuitive...).

Reply:
A better test would be to check that the elevation range before/after coregistration is more or less identical.

Reply:
Changed to use compressed() in 2d13111!

I also added tests to check that coregistration improved. I see now that I misread your suggestion haha, but this should be more than apt, no?

Reply:
Well, the overall statistics might not be too sensitive to just a few outliers. But the range will be, so I would still recommend checking the range.
But it's a good idea to check that the dispersion is reduced too!